.github/workflows/AddPRComment.yml

```diff
@@ -0,0 +1,24 @@
+on:
+  pull_request_target:
```

Comment:
Note: In public repositories this action does not work in pull_request workflows when triggered by forks.
Any attempt will be met with the error, resource not accessible by integration.
This is due to token restrictions put in place by GitHub Actions.
Private repositories can be configured to [enable workflows](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#enabling-workflows-for-forks-of-private-repositories) from forks to run without restriction.
See [here](https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#restrictions-on-repository-forks) for further explanation.
Alternatively, use the [pull_request_target](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target) event to comment on pull requests for public repo with forks.